### PR TITLE
Type fixes

### PR DESCRIPTION
**Pull request by datadanne**
Just a few small type-related improvements to `runContextualizers` and `helpers/utils`

### CODE REVIEW

File: src/commands/runContextualizers.ts [hunk 1]
```diff
@@ -20,46 +21,43 @@ export function registerRunContextualizersCommand() {
...
+              console.error(
+                `failed to run ${contextualizerName} on ${transaction.hash}: `,
+                err,
+              );
             }
-          });
+          }
+        });
 
-        await Promise.all(contextualizersPromise);
```
Comment (pcowgill) [inline src/commands/runContextualizers.ts:58]: @datadanne @ponyjackal What's the impact of this removal?

**Review by pcowgill — CHANGES_REQUESTED** [review]
@datadanne All the type changes look great! Before we merge, can you help me understand the removal of the `await Promise.all`?

File: src/commands/runContextualizers.ts [hunk 2]
```diff
@@ -20,46 +21,43 @@ export function registerRunContextualizersCommand() {
...
-                  err,
+                  `failed to run ${contextualizerName} on ${transaction.hash}`,
                 );
               }
+            } catch (err) {
+              console.error(err);
+            }
+          }
+          // run protocol contextualizers
+          for (const [contextualizerName, { contextualize }] of Object.entries(
```
Comment (pcowgill) [inline src/commands/runContextualizers.ts:42]: @ponyjackal This is a nice way to get access to the `contextualize` variable directly

File: src/commands/runContextualizers.ts [hunk 3]
```diff
@@ -20,46 +21,43 @@ export function registerRunContextualizersCommand() {
...
-          });
+          }
+        });
 
-        await Promise.all(contextualizersPromise);
         console.log('Successfully ran contextualizers');
         process.exit(0); // Successful exit
       } catch (error) {
-        console.error('Failed to grab the transaction:', error);
+        console.error('Running contextualizers failed:', error);
```
Comment (pcowgill) [inline src/commands/runContextualizers.ts:60]: Ah yes, good catch. Thanks!

File: src/commands/runContextualizers.ts (hunk 1 above)
Comment (datadanne) [reply src/commands/runContextualizers.ts:58]: No impact! The `contextualize` functions do not return promises, and the previous map over the `transactions` list did not return a list of promises (it didn't return anything, so the resulting value was of type `void[]`). awaiting void is a no-op

File: src/commands/runContextualizers.ts (hunk 2 above)
Comment (datadanne) [reply src/commands/runContextualizers.ts:42]: The nicest thing about this is is that tsc knows what type the `contextualize` value has here, it wasn't able to infer the type before when it was accessed using thearray notation with a non-const string value and therefore we didn't get any help from the compiler when using the value (it "was of type" any before)

File: src/helpers/utils.ts [hunk 4]
```diff
@@ -1,6 +1,10 @@
 import { utils } from 'ethers';
 import { InterfaceAbi } from '../types/Abi';
-import { TransactionContextType, Transaction } from '../types/transaction';
+import {
+  TransactionContextType,
+  Transaction,
```
[the remark this answers is not in the document]
Comment (pcowgill) [reply src/helpers/utils.ts:5]: I didn't see usage of these in the diff, so I was confused how it was working previously if these are needed

File: src/commands/runContextualizers.ts (hunk 1 above)
Comment (pcowgill) [reply src/commands/runContextualizers.ts:58]: @ponyjackal FYI ^

File: src/commands/runContextualizers.ts (hunk 2 above)
Comment (ponyjackal) [reply src/commands/runContextualizers.ts:42]: Yeah, but we do need to sacrifice readability, which is important in opensource project

File: src/commands/runContextualizers.ts (hunk 1 above)
Comment (ponyjackal) [reply src/commands/runContextualizers.ts:58]: Yeah, its a good find, I meant to run contextualizers in parallel when we relied on our database for logs, but no need now

File: src/commands/runContextualizers.ts (hunk 2 above)
Comment (pcowgill) [reply src/commands/runContextualizers.ts:42]: Maybe there's an option with the best of both worlds, automatic typing and readability?

Comment (pcowgill) [reply src/commands/runContextualizers.ts:42]: @ponyjackal Which specific aspect of this approach feels the least readable to you?

Comment (ponyjackal) [reply src/commands/runContextualizers.ts:42]:
```
 for (const [contextualizerName, { contextualize }] of Object.entries(protocolContextualizers))
```

Comment (ponyjackal) [reply src/commands/runContextualizers.ts:42]: Prob, we can use forEach to improve readability?

File: src/helpers/utils.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (datadanne) [reply src/helpers/utils.ts:5]: Ah I see. They were already imported, I just added `ContextSummaryVariableType`, and they are used in a part of this file that I didn't change which is why it doesn't show up in the diff

[the remark this answers is not in the document]
Comment (pcowgill) [reply src/helpers/utils.ts:5]: Oh haha I didn't read this closely enough, the indenting threw me off. Never mind!

File: src/commands/runContextualizers.ts (hunk 2 above)
Comment (datadanne) [reply src/commands/runContextualizers.ts:42]: Do you mean like this?
```ts
          Object.entries(protocolContextualizers).forEach(
            ([contextualizerName, { contextualize }]) => {
              ...
            }
          );
```
I personally prefer the version that is in the PR, but I'm happy to change it if you like this better!

Comment (pcowgill) [reply src/commands/runContextualizers.ts:42]: I'll tie-break here and say that the PR as-is is an improvement, so I'll merge it now, and we can think about any readability improvements we can make over time.

**Review by pcowgill — APPROVED** [review]

**Pull request merged**